internal/transport/http2_server.go

```diff
@@ -1032,55 +1042,38 @@ func (t *http2Server) deleteStream(s *Stream, eosReceived bool) {
 			atomic.AddInt64(&t.czData.streamsFailed, 1)
 		}
 	}
-}
-
-// closeStream clears the footprint of a stream when the stream is not needed
-// any more.
-func (t *http2Server) closeStream(s *Stream, rst bool, rstCode http2.ErrCode, hdr *headerFrame, eosReceived bool) {
-	// Mark the stream as done
-	oldState := s.swapState(streamDone)
 
-	// In case stream sending and receiving are invoked in separate
-	// goroutines (e.g., bi-directional streaming), cancel needs to be
-	// called to interrupt the potential blocking on other goroutines.
-	s.cancel()
-
-	// Deletes the stream from active streams
-	t.deleteStream(s, eosReceived)
+	return oldState
+}
 
-	cleanup := &cleanupStream{
-		streamID: s.id,
+// getCleanupStream returns a cleanupStream item that can be put into controlbuf.
+func getCleanupStream(streamID uint32, rst bool, rstCode http2.ErrCode) *cleanupStream {
```

Comment:
This function doesn't seem to be doing a lot.
Creating the struct inline should be good enough.